sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala

```diff
@@ -115,7 +115,8 @@ abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
      */
     private def canBroadcast(plan: LogicalPlan): Boolean = {
       plan.statistics.isBroadcastable ||
-        plan.statistics.sizeInBytes <= conf.autoBroadcastJoinThreshold
+        (plan.statistics.sizeInBytes >= 0 &&
```

Comment:
@cloud-fan @gatorsmile 
Thank you all for your comments.  I will change defaultSizeInBytes based on the above comments. 
Seems all the other places of Statistics.sizeInBytes are safe and will not cause overflow.